src/main/java/com/kodilla/ecommercee/controller/CartController.java

```
@@ -1,71 +1,48 @@
 package com.kodilla.ecommercee.controller;
 
 
-import com.kodilla.ecommercee.dto.CartDto;
-import org.springframework.http.MediaType;
-import org.springframework.http.ResponseEntity;
+import com.kodilla.ecommercee.dto.ProductDto;
+import com.kodilla.ecommercee.mapper.CartMapper;
+import com.kodilla.ecommercee.mapper.ProductMapper;
+import com.kodilla.ecommercee.service.CartService;
+import com.kodilla.ecommercee.service.UserService;
+import lombok.RequiredArgsConstructor;
 import org.springframework.web.bind.annotation.*;
 
-
-import java.math.BigDecimal;
-import java.util.ArrayList;
 import java.util.List;
 
+
+@RequiredArgsConstructor
 @RestController
+@RequestMapping("/cart")
 @CrossOrigin("*")
-@RequestMapping("/carts")
 public class CartController {
+    private final CartService cartDbService;
+    private final ProductMapper productMapper;
 
-    List<CartDto> CartDtoList = new ArrayList<CartDto>(){
-        {
-            add(new CartDto(1L, "kurtka zimowa", "Pellentesque tempus interdum quam ut rhoncus. Donec...", new BigDecimal(100), 1L));
-            add(new CartDto(2L, "płaszcz", "Pellentesque tempus interdum quam ut rhoncus. Donec...", new BigDecimal(150), 1L));
-            add(new CartDto(8L,"krawat", "Pellentesque tempus interdum quam ut rhoncus. Donec...", new BigDecimal(50), 2L));
-
-        }
-    };
 
-    @GetMapping
-    public ResponseEntity<List<CartDto>> getCart(){
-        return ResponseEntity.ok(CartDtoList);
+    @PostMapping("{userId}")
+    public void createEmptyCart(@PathVariable("userId") Long userId) {
+        cartDbService.createEmptyCart(userId);
     }
 
-    @GetMapping(value = "{cartId}")
-    public ResponseEntity<CartDto> getCart(@PathVariable Long cartId){
-        for (CartDto cart : CartDtoList) {
-            if (cart.getId().equals(cartId)){
-                return ResponseEntity.ok(cart);
-            }
-        }
-        return ResponseEntity.notFound().build();
+    @GetMapping("{userId}")
```

Maybe we should throw an exception here?

---

Yes, but subject of this task was Product entity and ProductController. I think this should be done in other pull - requests.

---

I don't know if it's worth creating a separate pull request for this. Tasks don't necessarily have to be 1-1 on PR. It's a team work tool and not something that is in any way important "outside". But It's your call...

---

Added exception throwing (UserNotFoundException)